pytoda/smiles/smiles_language.py

```diff
@@ -98,6 +99,10 @@ def load(filepath: str) -> 'SMILESLanguage':
         Returns:
             SMILESLanguage: the loaded SMILES language object.
         """
+        warnings.warn(
+            "Loading languages will use a text file in the future",
+            FutureWarning
+        )
```

Comment:
For the base smiles language,
```
sl = SMILESLanguage()
sl.load_vocab(vocab_file)
```
Does the trick.
Or are you concerned about SMILESTokenizer constructor arguments and want something like
```
@classmethod
def from_vocab(cls, vocab_file, **kwargs)
    sl = cls(**kwargs)
    sl.load_vocab(vocab_file)
    return sl
```

Reply:
Yep, the tokenizer was my concern. But if we build a tokenizer factory that stores a key in the json for the respective tokenizer, it should be fine. I don't see another nice way to store a regexp in a json than with a factory of tokenizers. This effectively prevents usecases of saving languages with individual tokenizers, but I think that's fine, given that the way how people tokenize SMILES seems to converge.

Reply:
we now have
```
TOKENIZER_FUNCTIONS: Dict[str, Tokenizer] = {
    'smiles': tokenize_smiles,
    'kmer_smiles': kmer_smiles_tokenizer,
    'spe_smiles': spe_smiles_tokenizer,
    'selfies': tokenize_selfies,
}
```
And these literals can be given to the SMILESLanguages as `tokenizer_name`. Is the behavior clear from this?
```
            smiles_tokenizer (Tokenizer): optional SMILES tokenization
                function. Defaults to tokenize_smiles, but tokenizer_name takes
                precedence when found in available TOKENIZER_FUNCTIONS.
            tokenizer_name (str): name, mapping to Tokenizer used to save and
                restore object from text files. Defaults to None, i.e.
                using default smiles_tokenizer.
```
This allows users to either pass a Tokenizer function directly via `smiles_tokenizer`, or to overload the `TOKENIZER_FUNCTIONS` for their projects like this:
```
In [1]: from pytoda.smiles import SMILESLanguage                                                                                                                                                                                               

In [2]: TOKENIZER_FUNCTIONS = {'new': list}                                                                                                                                                                                                    

In [3]: smiles_language = SMILESLanguage(tokenizer_name='new')                                                                                                                                                                                 

In [4]: smiles_language.add_smiles('C')                                                                                                                                                                                                        

In [5]: smiles_language.smiles_to_token_indexes('CCCCCC)                                                                                                                                                                                           
Out[5]: [35, 35, 35, 35, 35, 35]
```

I remember I had to set some defaults for kmer_smiles so to accept only a smiles. Now our definition is just a different default than the imported function. What do you think about not redefining it, but just importing and placing in the factory?

Reply:
Great, that's exactly what I had in mind. 
I would just add a list of currently supported tokenizers into the docstring of `tokenizer_name`

Reply:
Since the TOKENIZER_FUNCTIONS is intended to be overloaded, there cannot be a definite list of options.
Compared to before this actually gives a pointer to provided implementations. I think we can expect users to inspect TOKENIZER_FUNCTIONS.keys() themselves.

Reply:
I added some available names as you suggested.